controllers/horizon_controller.go

```diff
@@ -340,6 +344,21 @@ func (r *HorizonReconciler) reconcileNormal(ctx context.Context, instance *horiz
 		return ctrl.Result{}, err
 	}
 
+	memcachedExists, err := r.checkMemcachedCR(ctx, instance)
```

Comment:
I think this approach makes sense instead of always creating/patching the CR, because 
- this allows users to pre-create memcached CR
- this allows users to customize memcache CR(like scaling out) after it is auto-created
while this means we can't apply any update for auto-created memcached CR during update.

Alternatively we can create or patch memached cr unless user gives memcached cr name, and create as many replicas as horizon pod, but I don't know this can be a better approach than this.

Reply:
Using `controllerutil.CreateOrPatch()` now in the latest update.